bigchaindb/web/websocket_server.py

```diff
@@ -0,0 +1,173 @@
+"""WebSocket server for the BigchainDB Event Stream API."""
+
+# NOTE
+#
+# This module contains some functions and utilities that might belong to other
+# modules. For now, I prefer to keep everything in this module. Why? Because
+# those functions are needed only here.
+#
+# When we will extend this part of the project and we find that we need those
+# functionalities elsewhere, we can start creating new modules and organizing
+# things in a better way.
+
+
+import json
+import asyncio
+import logging
+import threading
+from uuid import uuid4
+
+import aiohttp
+from aiohttp import web
+
+from bigchaindb import config
+from bigchaindb.events import EventTypes
+
+
+logger = logging.getLogger(__name__)
+POISON_PILL = 'POISON_PILL'
+EVENTS_ENDPOINT = '/api/v1/streams/'
+
+
+def _put_into_capped_queue(queue, value):
+    """Put a new item in a capped queue.
+
+    If the queue reached its limit, get the first element
+    ready and put the new one. Note that the first element
+    will be lost (that's the purpose of a capped queue).
+
+    Args:
+        queue: a queue
+        value: the value to put
+    """
+    while True:
+        try:
+            queue.put_nowait(value)
+            return
+        except asyncio.QueueFull:
+            queue.get_nowait()
```

Comment:
A `try except else` pattern could be used here, for readability and style reasons, since obviously not much can go wrong with the `return` statement :).

```python
try:
    queue.put_nowait(value)
except asyncio.QueueFull:
    queue.get_nowait()
else:
    return
```